sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/AlterTableAddPartitionSuite.scala

```diff
@@ -124,18 +123,4 @@ class AlterTableAddPartitionSuite
       checkPartitions(t, Map("id" -> "1"), Map("id" -> "2"))
     }
   }
-
-  test("SPARK-40798: Alter partition should verify partition value - legacy") {
-    withNamespaceAndTable("ns", "tbl") { t =>
-      sql(s"CREATE TABLE $t (c int) $defaultUsing PARTITIONED BY (p int)")
-
-      withSQLConf(
-          SQLConf.SKIP_TYPE_VALIDATION_ON_ALTER_PARTITION.key -> "true",
-          SQLConf.ANSI_ENABLED.key -> "false") {
```

Comment:
Shouldn't we maybe keep the test after removing `SQLConf.ANSI_ENABLED.key -> "false"`?

Reply:
Before I thought the `storeAssignmentPolicy` is enough so this test is corvered by `AlterTableAddPartitionSuiteBase`. Just change to legacy to restroe previous behavior.

Now I add one more `SKIP_TYPE_VALIDATION_ON_ALTER_PARTITION` check and keep this test.